src/parser/core/modules/Medicated.ts

```diff
@@ -43,22 +43,28 @@ export class Medicated extends Analyser {
 			throw new Error('potion instance not found')
 		}
 
-		this.pots.push({end: event.timestamp, ...this.pot})
+		this.pots.push({ ...this.pot, end: event.timestamp })
 
 		this.pot = undefined
 	}
 
 	private onComplete() {
 		const status = this.data.statuses.MEDICATED
 
+		// Add a pot that is still up before end of pull
+		if (this.pot && !this.pot.end) {
+			this.pots.push(this.pot)
+			this.pot = undefined
+		}
+
 		if (this.pots.length > 0) {
 			const row = new SimpleRow({
 				label: 'Medicated',
 				order: -99,
 			})
 
 			this.pots.forEach(pot => row.addItem(new StatusItem({
-				status: status,
+				status,
```

Comment:
Also why?

Reply:
Sorry this one's a bit more pedantic. I think its its just easier to read a deferred object creation that's closer to where its used.

Reply:
I find it harder to read this way, destructuring isn't complex but it is arcane.